File: app/src/main/java/com/tasomaniac/devdrawer/data/AppDao.java
package com.tasomaniac.devdrawer.data;

import android.arch.persistence.room.Dao;
import android.arch.persistence.room.Insert;
import android.arch.persistence.room.OnConflictStrategy;
import android.arch.persistence.room.Query;

import java.util.List;

@Dao
public interface AppDao {

  @Query("SELECT DISTINCT packageName FROM app WHERE appWidgetId = :appWidgetId")
  List<String> findAppsByWidgetIdSync(int appWidgetId);

  @Insert(onConflict = OnConflictStrategy.IGNORE)
  void insertAppSync(List<App> app);

  @Query("DELETE FROM app WHERE packageName = :packageName")
  void deleteAppSync(String packageName);

  @Query("DELETE FROM app WHERE packageMatcher = :packageMatcher")
  void deleteAppsByPackageMatcherSync(String packageMatcher);

  @Query("UPDATE app SET appWidgetId = :appWidgetId WHERE appWidgetId = 2147483647")
  void updateTempWidgetIdSync(int appWidgetId);
}

File: app/src/main/java/com/tasomaniac/devwidget/data/AppDao.kt
package com.tasomaniac.devwidget.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query

@Dao
interface AppDao {

    @Query("SELECT DISTINCT packageName FROM app WHERE appWidgetId = :appWidgetId")
    fun findAppsByWidgetIdSync(appWidgetId: Int): List<String>

    @Insert(onConflict = OnConflictStrategy.IGNORE)
    fun insertAppSync(app: List<App>)

    @Query("DELETE FROM app WHERE packageName = :packageName")
    fun deleteAppSync(packageName: String)

    @Query("DELETE FROM app WHERE packageMatcher = :packageMatcher")
    fun deleteAppsByPackageMatcherSync(packageMatcher: String)

}
